diff --git a/CMakeLists.txt b/CMakeLists.txt
index 52a3f05..e070825 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,9 +31,9 @@ set(Boost_NO_BOOST_CMAKE ON)
 #find_package(PythonLibs ${PY_VERSION} REQUIRED)
 find_package(PythonInterp REQUIRED)
 #find_package(PythonLibs ${PYTHON_VERSION_STRING} REQUIRED)
-find_package(PythonLibs 2.7 REQUIRED)
+find_package(PythonLibs 3 REQUIRED)
 #message(STATUS "Using python version ${PYTHON_VERSION_STRING}")
-find_package(Boost REQUIRED COMPONENTS date_time filesystem python signals system)
+find_package(Boost REQUIRED COMPONENTS date_time filesystem python37 signals system)
 #find_package(Boost REQUIRED COMPONENTS date_time filesystem python-py36 signals system)
 find_library(FFTW3_LIB fftw3 REQUIRED)
 enable_language(Fortran OPTIONAL)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d6739e3..ab6ba08 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -236,7 +236,7 @@ set(ALL_LIBRARIES
 	${Boost_SYSTEM_LIBRARY}
 	${Boost_FILESYSTEM_LIBRARY} ${Boost_DATE_TIME_LIBRARY}
 #	${Boost_PYTHON-PY36_LIBRARIES}
-	${Boost_PYTHON_LIBRARIES}
+	${Boost_PYTHON37_LIBRARIES}
 	${Boost_SIGNALS_LIBRARY}
 	${FFTW3_LIB}
 	${CASACORE_LIBRARIES}
