--- gcwrap/tools/CMakeLists.txt.orig	2016-04-19 11:12:54.323848825 -0400
+++ gcwrap/tools/CMakeLists.txt	2016-04-19 11:21:09.921110755 -0400
@@ -39,26 +39,8 @@
 add_custom_target(
     tools_python_casadef
     COMMAND echo "import os" > casadef.py
-    COMMAND echo "task_directory = None" >> casadef.py
-    COMMAND echo "\\#\\# CASAPATH is the primary environment variable" >> casadef.py
-    COMMAND echo "if os.environ.has_key\\\(\\'CASAPATH\\'\\\):" >> casadef.py
-    COMMAND echo "    __casapath__ = os.environ[\\'CASAPATH\\'].split\\\(\\' \\'\\\)[0]" >> casadef.py
-    COMMAND echo "    __casaarch__ = os.environ[\\'CASAPATH\\'].split\\\(\\' \\'\\\)[1]" >> casadef.py
-    COMMAND echo "    if os.path.exists\\\(__casapath__ + \\'/lib/python2.7/casapy.py\\'\\\):" >> casadef.py
-    COMMAND echo "        task_directory = __casapath__ + \\'/lib/python2.7\\'" >> casadef.py
-    COMMAND echo "        python_library_directory = task_directory" >> casadef.py
-    COMMAND echo "    elif os.path.exists\\\(__casapath__ + \\'/Resources/python/casapy.py\\'\\\):" >> casadef.py
-    COMMAND echo "        task_directory = __casapath__ + \\'/Resources/python\\'" >> casadef.py
-    COMMAND echo "        python_library_directory = task_directory" >> casadef.py
-    COMMAND echo "    elif os.path.exists\\\(__casapath__ + \\'/\\' + __casaarch__ + \\'/python/2.7/casapy.py\\'\\\):" >> casadef.py
-    COMMAND echo "        task_directory = __casapath__ + \\'/\\' + __casaarch__ + \\'/python/2.7\\'" >> casadef.py
-    COMMAND echo "        python_library_directory = task_directory" >> casadef.py
-    COMMAND echo "\\#\\# if CASAPATH does not yeild any result, check secondary variables..." >> casadef.py
-    COMMAND echo "if task_directory is None:" >> casadef.py
-    COMMAND echo "    if os.environ.has_key\\\(\\'__CASAPY_TASKDIR\\'\\\):" >> casadef.py
-    COMMAND echo "        task_directory = os.environ[\\'__CASAPY_TASKDIR\\']" >> casadef.py
-    COMMAND echo "    if os.environ.has_key\\\(\\'__CASAPY_PYTHONDIR\\'\\\):" >> casadef.py
-    COMMAND echo "        python_library_directory = os.environ[\\'__CASAPY_PYTHONDIR\\']" >> casadef.py
+    COMMAND echo "task_directory = os.environ.get\\\(\\'__CASAPY_TASKDIR\\',\\'${CMAKE_INSTALL_PREFIX}/lib/casa/tasks\\'\\\)" >> casadef.py
+    COMMAND echo "python_library_directory = os.environ.get\\\(\\'__CASAPY_PYTHONDIR\\',task_directory\\\)" >> casadef.py
     COMMAND echo "casa_version = \\\"${CASA_MAJOR_VERSION}.${CASA_MINOR_VERSION}.${CASA_PATCH_VERSION}\\\"" >> casadef.py
     COMMAND cat definitions >> casadef.py
     DEPENDS definitions
