--- gcwrap/CMakeLists.txt.orig	2016-06-27 03:54:40.063271385 +0000
+++ gcwrap/CMakeLists.txt	2016-06-27 03:54:48.001270578 +0000
@@ -1204,7 +1204,7 @@
 casa_find( READLINE
   VERSION 4.3
   INCLUDES readline/readline.h #depends on <cstdio> to compile
-  LIBS readline ncurses
+  LIBS readline ncursesw
   PREFIX_HINTS ${READLINE_ROOT_DIR}
   CPP_VERSION "RL_VERSION_MAJOR << '.' << RL_VERSION_MINOR"
   RUN_VERSION rl_library_version
