--- make.include.orig	2017-02-06 22:46:09.641647497 -0500
+++ make.include	2017-02-06 22:48:21.068855178 -0500
@@ -58,7 +58,7 @@
 	./configure \
 	$(WITHTK) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
-	--enable-symbols
+	--enable-symbols $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tclcheckdns
 
@@ -75,7 +75,7 @@
 	$(WITHTCL) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tcliis
 
@@ -92,7 +92,7 @@
 	$(WITHTCL) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tclsignal
 
@@ -109,7 +109,7 @@
 	$(WITHTCL) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tclxml
 
@@ -128,7 +128,7 @@
 	$(TCLXMLFLAGS) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tclzvfs
 
@@ -145,7 +145,7 @@
 	$(WITHTCL) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- xpa
 
@@ -164,7 +164,7 @@
 	$(WITHTCL) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	--with-x=disabled
+	--with-x=disabled $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tkblt
 
@@ -181,7 +181,7 @@
 	$(WITHTK) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tkcon
 
@@ -208,7 +208,7 @@
 	$(WITHTK) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tkimg
 
@@ -223,10 +223,10 @@
 	@echo "*** Configure tkimg ***"
 	cd tkimg; \
 	./configure \
-	CFLAGS='-DPNG_NO_WRITE_gAMA' \
+	CFLAGS="-DPNG_NO_WRITE_gAMA $$CFLAGS" \
 	$(WITHTK) $(TKIMGFLAGS) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
-	--disable-shared --enable-symbols
+	--disable-shared --enable-symbols $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tkmpeg
 
@@ -243,7 +243,7 @@
 	$(WITHTK) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tksao
 
@@ -260,7 +260,7 @@
 	$(WITHTK) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- ast
 
@@ -276,7 +276,7 @@
 	star_cv_cnf_trail_type=long star_cv_cnf_f2c_compatible=no \
 	$(CONFIGFLAGS) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
-	--disable-shared
+	--disable-shared $$EXTRA_CONFIGURE_ARGS
 
 ast/ast.h :
 	@echo ""
@@ -335,7 +335,7 @@
 	--with-zlib=../lib --with-wcslib=../lib \
 	$(CONFIGFLAGS) $(TARGET) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
-	--disable-shared
+	--disable-shared $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- tktable
 
@@ -353,7 +353,7 @@
 	$(WITHTK) \
 	--prefix $(ROOT) --exec-prefix $(ROOT) \
 	--disable-shared --enable-symbols \
-	$(CACHE)
+	$(CACHE) $$EXTRA_CONFIGURE_ARGS
 
 #-------------------------- language
 
