--- Makefile.in.orig	2015-04-20 21:30:41.644913712 +0000
+++ Makefile.in	2015-04-20 21:30:48.510906275 +0000
@@ -493,7 +493,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = dbus bus tools test doc
+SUBDIRS = dbus tools
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = dbus-1.pc
 DISTCLEANFILES = \
--- configure.orig	2015-04-20 21:29:50.860968721 +0000
+++ configure	2015-04-20 21:31:34.379856590 +0000
@@ -19400,62 +19400,6 @@
 
 #### Sort out XML library
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for XML_ParserCreate_MM in -lexpat" >&5
-$as_echo_n "checking for XML_ParserCreate_MM in -lexpat... " >&6; }
-if ${ac_cv_lib_expat_XML_ParserCreate_MM+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lexpat  $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char XML_ParserCreate_MM ();
-int
-main ()
-{
-return XML_ParserCreate_MM ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_expat_XML_ParserCreate_MM=yes
-else
-  ac_cv_lib_expat_XML_ParserCreate_MM=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_expat_XML_ParserCreate_MM" >&5
-$as_echo "$ac_cv_lib_expat_XML_ParserCreate_MM" >&6; }
-if test "x$ac_cv_lib_expat_XML_ParserCreate_MM" = xyes; then :
-   for ac_header in expat.h
-do :
-  ac_fn_c_check_header_mongrel "$LINENO" "expat.h" "ac_cv_header_expat_h" "$ac_includes_default"
-if test "x$ac_cv_header_expat_h" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_EXPAT_H 1
-_ACEOF
-
-else
-  as_fn_error $? "Could not find expat.h, check config.log for failed attempts" "$LINENO" 5
-fi
-
-done
-
-else
-   as_fn_error $? "Explicitly requested expat but expat not found" "$LINENO" 5
-fi
-
-
 XML_LIBS=-lexpat
 XML_CFLAGS=
 
