diff --git a/src/gui/kernel/qcursor_mac.mm b/src/gui/kernel/qcursor_mac.mm
index c8ccf75..60f5b03 100644
--- a/src/gui/kernel/qcursor_mac.mm
+++ b/src/gui/kernel/qcursor_mac.mm
@@ -319,7 +319,7 @@ void qt_mac_update_cursor_at_global_pos(const QPoint &globalPos)
     pos.x = x;
     pos.y = y;
 
-    CGEventRef e = CGEventCreateMouseEvent(0, kCGEventMouseMoved, pos, 0);
+    CGEventRef e = CGEventCreateMouseEvent(0, kCGEventMouseMoved, pos, (CGMouseButton) 0);
     CGEventPost(kCGHIDEventTap, e);
     CFRelease(e);
 #else

diff --git a/src/gui/kernel/qwidget_mac.mm b/src/gui/kernel/qwidget_mac.mm
index 5730020..b31bf0d 100644
--- a/src/gui/kernel/qwidget_mac.mm
+++ b/src/gui/kernel/qwidget_mac.mm
@@ -3620,7 +3620,7 @@ QPoint qt_mac_nativeMapFromParent(const QWidget *child, const QPoint &pt)
 #else
             // Only needed if it exists from 10.7 or later
             if ((q->windowType() == Qt::Tool) && [window respondsToSelector: @selector(setAnimationBehavior:)])
-                [window setAnimationBehavior: 2]; // NSWindowAnimationBehaviorNone == 2
+                [window setAnimationBehavior: (NSWindowAnimationBehavior) 2]; // NSWindowAnimationBehaviorNone == 2
 
             [window orderOut:window];
             // Unfortunately it is not as easy as just hiding the window, we need

diff --git a/src/gui/kernel/qcocoamenuloader_mac.mm b/src/gui/kernel/qcocoamenuloader_mac.mm
index ca2481f..c4cd65f 100644
--- a/src/gui/kernel/qcocoamenuloader_mac.mm
+++ b/src/gui/kernel/qcocoamenuloader_mac.mm
@@ -125,7 +125,7 @@ - (void)ensureAppMenuInMenu:(NSMenu *)menu
 - (void)removeActionsFromAppMenu
 {
     for (NSMenuItem *item in [appMenu itemArray])
-        [item setTag:nil];
+        [item setTag: (NSInteger) nil];
 }
 
 - (void)dealloc

diff --git a/src/gui/text/qfontengine_coretext.mm b/src/gui/text/qfontengine_coretext.mm
index 204d685..e05179e 100644
--- a/src/gui/text/qfontengine_coretext.mm
+++ b/src/gui/text/qfontengine_coretext.mm
@@ -886,7 +886,7 @@ static void convertCGPathToQPainterPath(void *info, const CGPathElement *element
 
 QFixed QCoreTextFontEngine::emSquareSize() const
 {
-    return QFixed::QFixed(int(CTFontGetUnitsPerEm(ctfont)));
+    return QFixed(int(CTFontGetUnitsPerEm(ctfont)));
 }
 
 QFontEngine *QCoreTextFontEngine::cloneWithSize(qreal pixelSize) const

diff --git a/src/gui/widgets/qmenu_mac.mm b/src/gui/widgets/qmenu_mac.mm
index 6a9cbc7..8692714 100644
--- a/src/gui/widgets/qmenu_mac.mm
+++ b/src/gui/widgets/qmenu_mac.mm
@@ -769,7 +769,7 @@ bool qt_mac_menubar_is_open()
                    && menuItem != [getMenuLoader() quitMenuItem]) {
             [menuItem setHidden:YES];
         }
-        [menuItem setTag:nil];
+        [menuItem setTag: (NSInteger) nil];
     }
     [menuItem release];
 #endif

