--- mkspecs/unsupported/macx-clang-libc++/qmake.conf.orig	2016-12-27 12:51:31.000000000 +1000
+++ mkspecs/unsupported/macx-clang-libc++/qmake.conf	2016-12-27 12:52:13.000000000 +1000
@@ -13,7 +13,7 @@
 include(../../common/gcc-base-macx.conf)
 include(../../common/clang.conf)
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7 # Libc++ is available from 10.7 onwards
+QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.9 # conda-forge minimum
 
 QMAKE_CFLAGS += -mmacosx-version-min=$$QMAKE_MACOSX_DEPLOYMENT_TARGET
 QMAKE_CXXFLAGS += -stdlib=libc++ -mmacosx-version-min=$$QMAKE_MACOSX_DEPLOYMENT_TARGET
