--- a/src/gui/painting/qpaintengine_mac.cpp	2015-05-07 07:14:43.000000000 -0700
+++ b/src/gui/painting/qpaintengine_mac.cpp	2015-07-06 18:07:52.000000000 -0700
@@ -340,13 +340,7 @@
     }
 
     // Get the color space from the display profile.
-    CGColorSpaceRef colorSpace = 0;
-    CMProfileRef displayProfile = 0;
-    CMError err = CMGetProfileByAVID((CMDisplayIDType)displayID, &displayProfile);
-    if (err == noErr) {
-        colorSpace = CGColorSpaceCreateWithPlatformColorSpace(displayProfile);
-        CMCloseProfile(displayProfile);
-    }
+    CGColorSpaceRef colorSpace = CGDisplayCopyColorSpace(displayID);
 
     // Fallback: use generic DeviceRGB
     if (colorSpace == 0)
