--- asap/cmake/withcasa.cmake.orig	2015-05-09 17:02:44.830146159 +0000
+++ asap/cmake/withcasa.cmake	2015-05-09 17:15:40.486284857 +0000
@@ -110,22 +110,11 @@
 #
 # casacore
 #
-set( CASACORE_PATHS "${casaroot}/${arch};${casaroot};/usr/local;/usr" )
+set( CASACORE_PATHS ${CMAKE_INSTALL_PREFIX} )
 
 SET(NO_SOVERSION FALSE CACHE BOOL "do not add version information to shared libraries")
 if( NOT NO_SOVERSION )
-    set( epochdelta 1385403204 )
-    if ( EXISTS ${CMAKE_INSTALL_PREFIX}/${arch}/casa_sover.txt )
-        execute_process( COMMAND perl -e "while (<>) { chomp and print if (! m/^\#/ ) }" ${CMAKE_INSTALL_PREFIX}/${arch}/casa_sover.txt
-                         OUTPUT_VARIABLE __asap_soversion )
-    elseif( EXISTS ${CMAKE_INSTALL_PREFIX}/casa_sover.txt )
-        execute_process( COMMAND perl -e "while (<>) { chomp and print if (! m/^#/ ) }" ${CMAKE_INSTALL_PREFIX}/casa_sover.txt
-                         OUTPUT_VARIABLE __asap_soversion )
-    else( )
-        execute_process( COMMAND perl -e "$t=time( )-${epochdelta};$z=$t & 0xff; $y=($t>>8)&0xff; $x=($t>>16)&0xffff; print \"$x.$y.$z\""
-                         OUTPUT_VARIABLE __asap_soversion )
-    endif( )
-    set(asap_soversion ${__asap_soversion} CACHE STRING "version for shared objects")
+    set(asap_soversion "46.0.0" CACHE STRING "version for shared objects")
     message( STATUS "Shared object version number ${asap_soversion}" )
 else( )
     message( STATUS "User disabled shared library versioning" )
