diff --git a/tectonic/XeTeX_ext.c b/tectonic/XeTeX_ext.c
index 6175eb5..7486da4 100644
--- a/tectonic/XeTeX_ext.c
+++ b/tectonic/XeTeX_ext.c
@@ -36,6 +36,8 @@ authorization from the copyright holders.
  * additional plain C extensions for XeTeX - mostly platform-neutral
  */
 
+#include <zlib.h> /* must include before TECkit_Common.h due to a redundant Byte typedef */
+
 #include <tectonic/tectonic.h>
 #include <tectonic/internals.h>
 #include <tectonic/XeTeX_ext.h>
@@ -57,7 +59,6 @@ authorization from the copyright holders.
 
 #include <poppler-config.h>
 #include <png.h>
-#include <zlib.h>
 #include <graphite2/Font.h>
 
 #include "xetexd.h"
diff --git a/tectonic/synctex.c b/tectonic/synctex.c
index d0d24a8..8365ce2 100644
--- a/tectonic/synctex.c
+++ b/tectonic/synctex.c
@@ -122,6 +122,8 @@
 
  */
 
+#include <zlib.h> /* must include before TECkit_Common.h due to a redundant Byte typedef */
+
 #include <tectonic/tectonic.h>
 #include <tectonic/internals.h>
 #include <tectonic/xetexd.h>
@@ -129,7 +131,6 @@
 
 #include <stdio.h>
 #include <stdarg.h>
-#include <zlib.h>
 
 #define SYNCTEX_VERSION 1
 #define SYNCTEX_DEBUG 0
