diff --git a/deconvolution/peakfinder.cpp b/deconvolution/peakfinder.cpp
index 5476c49..e4dc906 100644
--- a/deconvolution/peakfinder.cpp
+++ b/deconvolution/peakfinder.cpp
@@ -9,6 +9,7 @@
 #include <immintrin.h>
 #endif

+#include <algorithm>
 #include <limits>

 std::optional<float> PeakFinder::Simple(const float *image, size_t width,
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2536355..bcbb29f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -415,7 +415,6 @@ add_executable(
 target_link_libraries(mscaleexample ${ALL_LIBRARIES})
 
 install(TARGETS wsclean DESTINATION bin)
-install(TARGETS wsclean-lib DESTINATION lib)
 install(TARGETS chgcentre DESTINATION bin)
 install(FILES interface/wscleaninterface.h DESTINATION include)
 
