--- configure.orig	2017-04-14 10:30:50.783073809 -0400
+++ configure	2017-04-14 10:32:21.101565524 -0400
@@ -3007,7 +3007,10 @@
 else
     pytmp="${top}/ots/bin"
 fi
-if test -e ${pytmp}/../lib/python3.5
+if test -e ${pytmp}/../lib/python3.6
+then
+    pyver="3.6"
+elif test -e ${pytmp}/../lib/python3.5
 then
     pyver="3.5"
 elif test -e ${pytmp}/../lib/python2.7
@@ -3025,7 +3028,13 @@
 fi
 
 
-if test "${pyver}" = "3.5"
+if test "${pyver}" = "3.6"
+then
+    pyminver="6"
+    pymajver="3"
+    pyinc="python${pyver}m"
+    pythonlib="-L\$(PYTHON)/lib -lpython\$(PYTHONVER)m"
+elif test "${pyver}" = "3.5"
 then
     pyminver="5"
     pymajver="3"
@@ -10381,10 +10390,22 @@
     then
         ciao_setup="ots/lib/python3.5/config/Makefile $ciao_setup"
     fi
+    if test -f "ots/lib/python3.6/config-3.6m/Makefile.in"
+    then
+        ciao_setup="ots/lib/python3.6/config-3.6m/Makefile $ciao_setup"
+    fi
+    if test -f "ots/lib/python3.6/config/Makefile.in"
+    then
+        ciao_setup="ots/lib/python3.6/config/Makefile $ciao_setup"
+    fi
     if test -f "ots/lib/python2.7/_sysconfigdata.py.in"
     then
         ciao_setup="ots/lib/python2.7/_sysconfigdata.py $ciao_setup"
     fi
+    if test -f "ots/lib/python3.6/_sysconfigdata.py.in"
+    then
+        ciao_setup="ots/lib/python3.6/_sysconfigdata.py $ciao_setup"
+    fi
     if test -f "ots/lib/python3.5/_sysconfigdata.py.in"
     then
         ciao_setup="ots/lib/python3.5/_sysconfigdata.py $ciao_setup"
@@ -10624,6 +10645,14 @@
    then
       startup="ots/lib/python3.5/config-3.5m/Makefile $startup"
    fi
+   if test -f "ots/lib/python3.6/config/Makefile.in"
+   then
+      startup="ots/lib/python3.6/config/Makefile $startup"
+   fi
+   if test -f "ots/lib/python3.6/config-3.6m/Makefile.in"
+   then
+      startup="ots/lib/python3.6/config-3.6m/Makefile $startup"
+   fi
    if test -f "$user_bin/ciao-python-fix.in"
    then
      startup="$startup $user_bin/ciao-python-fix"
--- src/libdev/grplib/configure.orig	2017-04-14 10:34:34.582292226 -0400
+++ src/libdev/grplib/configure	2017-04-14 10:35:27.419579885 -0400
@@ -10883,26 +10883,6 @@
 	#
 	# if the macro parameter ``version'' is set, honour it
 	#
-	if test -n "<'3.5.2'"; then
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for a version of Python <'3.5.2'" >&5
-$as_echo_n "checking for a version of Python <'3.5.2'... " >&6; }
-		ac_supports_python_ver=`$PYTHON -c "import sys; \
-			ver = sys.version.split ()[0]; \
-			print (ver <'3.5.2')"`
-		if test "$ac_supports_python_ver" = "True"; then
-		   { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-		else
-			{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-			as_fn_error "this package requires Python <'3.5.2'.
-If you have it installed, but it isn't the default Python
-interpreter in your system path, please pass the PYTHON_VERSION
-variable to configure. See \`\`configure --help'' for reference.
-" "$LINENO" 5
-			PYTHON_VERSION=""
-		fi
-	fi
 
 	#
 	# Check if you have distutils, else fail
