--- configure.orig	2016-03-23 22:16:07.499980260 +0000
+++ configure	2016-03-23 22:16:54.440128838 +0000
@@ -18716,7 +18716,7 @@
 
 # These variables are only ever used when we cross-build to X86_WIN32.
 # And we only support this with GCC, so...
-if test "x$GCC" = "xyes"; then
+if false; then
   if test -n "$with_cross_host" &&
      test x"$with_cross_host" != x"no"; then
     toolexecdir="${exec_prefix}"/'$(target_alias)'
--- Makefile.in.orig	2016-03-23 22:16:12.395995757 +0000
+++ Makefile.in	2016-03-23 22:20:02.501724101 +0000
@@ -698,7 +698,7 @@
 # Subdir rules rely on $(FLAGS_TO_PASS)
 FLAGS_TO_PASS = $(AM_MAKEFLAGS)
 MAKEOVERRIDES = 
-toolexeclib_LTLIBRARIES = libffi.la
+toolexeclib_LTLIBRARIES = libffi32.la
 noinst_LTLIBRARIES = libffi_convenience.la
 libffi_la_SOURCES = src/prep_cif.c src/types.c \
 		src/raw_api.c src/java_raw_api.c src/closures.c
@@ -1161,7 +1161,7 @@
 src/vax/ffi.lo: src/vax/$(am__dirstamp) \
 	src/vax/$(DEPDIR)/$(am__dirstamp)
 
-libffi.la: $(libffi_la_OBJECTS) $(libffi_la_DEPENDENCIES) $(EXTRA_libffi_la_DEPENDENCIES) 
+libffi32.la: $(libffi_la_OBJECTS) $(libffi_la_DEPENDENCIES) $(EXTRA_libffi_la_DEPENDENCIES) 
 	$(AM_V_CCLD)$(libffi_la_LINK) -rpath $(toolexeclibdir) $(libffi_la_OBJECTS) $(libffi_la_LIBADD) $(LIBS)
 
 libffi_convenience.la: $(libffi_convenience_la_OBJECTS) $(libffi_convenience_la_DEPENDENCIES) $(EXTRA_libffi_convenience_la_DEPENDENCIES) 
