--- code/CMakeLists.txt.orig	2015-05-13 21:10:10.230512951 +0000
+++ code/CMakeLists.txt	2015-05-13 21:10:58.967612985 +0000
@@ -384,7 +384,7 @@
 	set( casaroot ${casaroot} CACHE PATH "toplevel casa directory" FORCE )
     else()
         set( CMAKE_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX} CACHE PATH "casa architecture directory" FORCE )
-	set( casaroot ${CMAKE_INSTALL_PREFIX}/.. CACHE PATH "toplevel casa directory" FORCE )
+	set( casaroot ${CMAKE_INSTALL_PREFIX}/lib/casa CACHE PATH "toplevel casa directory" FORCE )
    endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 endif()
 
