--- code/casadbus/utilities/Conversion.cc.orig	2015-12-21 17:04:08.000000000 -0800
+++ code/casadbus/utilities/Conversion.cc	2015-12-21 17:07:12.000000000 -0800
@@ -57,7 +57,7 @@
 	    } else if ( src.signature( ) == "a{sv}" ) {
 		DBus::MessageIter ri = src.reader( );
 		std::map<std::string,DBus::Variant> vm;
-		::operator >>(ri,vm);
+		ri >> vm;
 		record rec;
 		for ( std::map<std::string,DBus::Variant>::iterator iter = vm.begin( );
 		      iter != vm.end( ); ++iter ) {
@@ -71,35 +71,35 @@
 	static void fromVariant( ::DBus::MessageIter &vi, const variant &src ) {
 	    switch ( src.type() ) {
 	    case variant::INT:
-		::operator <<(vi,src.getInt( ));
+		vi << src.getInt( );
 		break;
 	    case variant::BOOL:
-		::operator <<(vi,src.getBool( ));
+		vi << src.getBool( );
 		break;
 	    case variant::STRING:
-		::operator <<(vi,src.getString( ));
+		vi << src.getString( );
 		break;
 	    case variant::DOUBLE:
-		::operator <<(vi,src.getDouble( ));
+		vi << src.getDouble( );
 		break;
 	    case variant::BOOLVEC:
-		::operator <<(vi,src.getBoolVec( ));
+		vi << src.getBoolVec( );
 		break;
 	    case variant::INTVEC:
-		::operator <<(vi,src.getIntVec( ));
+		vi << src.getIntVec( );
 		break;
 	    case variant::DOUBLEVEC:
-		::operator <<(vi,src.getDoubleVec( ));
+		vi << src.getDoubleVec( );
 		break;
 	    case variant::STRINGVEC:
-		::operator <<(vi,src.getStringVec( ));
+		vi << src.getStringVec( );
 		break;
 	    case variant::RECORD:
 		{   const record &rec = src.getRecord( );
 		    ::DBus::MessageIter ait = vi.new_array("{sv}");
 		    for ( record::const_iterator rit = rec.begin( ); rit != rec.end( ); ++rit ) {
 		        ::DBus::MessageIter eit = ait.new_dict_entry( );
-			::operator <<(eit,rit->first);
+		        eit << rit->first;
 			::DBus::MessageIter val = eit.new_variant( rit->second.sig() );
 			fromVariant( val, rit->second );
 			eit.close_container(val);
--- code/synthesis/MeasurementEquations/Imager.cc~	2015-12-19 23:14:03.000000000 -0800
+++ code/synthesis/MeasurementEquations/Imager.cc	2015-12-21 19:32:42.000000000 -0800
@@ -7415,7 +7415,7 @@
 bool interactive_clean_callback::callback( const DBus::Message &msg ) {
     if (msg.is_signal("edu.nrao.casa.viewer","interact")) {
 	DBus::MessageIter ri = msg.reader( );
-	::operator >>(ri,result_);
+	ri >> result_;
 	casa::DBusSession::instance( ).dispatcher( ).leave( );
     }
     return true;
--- code/synthesis/ImagerObjects/InteractiveMasking.cc.orig	2015-12-21 19:39:55.000000000 -0800
+++ code/synthesis/ImagerObjects/InteractiveMasking.cc	2015-12-21 19:40:11.000000000 -0800
@@ -53,7 +53,7 @@
   bool new_interactive_clean_callback::callback( const DBus::Message &msg ) {
     if (msg.is_signal("edu.nrao.casa.viewer","interact")) {
       DBus::MessageIter ri = msg.reader( );
-      ::operator >>(ri,result_);
+      ri >> result_;
       casa::DBusSession::instance( ).dispatcher( ).leave( );
     }
     return true;
